hadrons/numba_files/continuous_beam_numba.py

```diff
@@ -0,0 +1,307 @@
+    no_z = int(d_cm/unit_length_cm) #number of elements in the z direction
+    no_z_electrode = 5 #length of the electrode-buffer to ensure no ions drift through the array in one time step
+    no_z_with_buffer = 2*no_z_electrode + no_z
+    # depending on the cluster/computer, the upper limit may be changed
+    if (no_xy*no_xy*no_z) > 1e8:
+        raise ValueError("Too many elements in the array: %i" % (no_xy*no_xy*no_z_with_buffer))
+
+    # find the middle of the arrays
+    mid_xy_array = int(no_xy/2.)
+    mid_z_array = int(no_z/2.)
+    outer_radius = (no_xy/2.)
+    inner_radius = outer_radius - buffer_radius
+
+    no_figure_updates = 5              # update the figure this number of times
+    number_of_iterations = 1e7      # preallocate track coordinates
+
+    '''
+    find a time step dt which fulfils the von Neumann criterion, i.e. ensures the numericl error does not increase but
+    decreases and eventually damps out
+    '''
+    dt = 1.
+    von_neumann_expression = False
+    Efield_V_cm = voltage_V/d_cm # to ensure the same dt in all simulations
+    while not von_neumann_expression:
+        dt /= 1.01
+        # as defined in the Deghan (2004) paper
+        sx = ion_diff*dt/(unit_length_cm**2)
+        sy = ion_diff*dt/(unit_length_cm**2)
+        sz = ion_diff*dt/(unit_length_cm**2)
+        cx = cy = 0.0
+        cz = ion_mobility*Efield_V_cm*dt/unit_length_cm
+        # check von Neumann's criterion
+        von_neumann_expression = (2*(sx + sy + sz) + cx**2 + cy**2 + cz**2 <= 1 and cx**2*cy**2*cz**2 <= 8*sx*sy*sz)
+
+    '''
+    Calculate the number of step required to drag the two charge carrier distributions apart
+    along with the number of tracks to be uniformly distributed over the domain
+    '''
+    separation_time_steps = int(d_cm/(2.*ion_mobility*Efield_V_cm*dt))
+    n_separation_times = 3
+    computation_time_steps = separation_time_steps * n_separation_times
+    simulation_time_s = computation_time_steps*dt
+    number_of_tracks = int(fluence_cm2_s* simulation_time_s*area_cm2)
+    if number_of_tracks < 1:
+        number_of_tracks = 1 # initial recombination
+
+    # Gaussian track structure
+    N0 = LET_eV_cm/W                 # Linear charge carrier density
+    b_cm = track_radius              # Gaussian track radius
+    Gaussian_factor = N0/(pi*b_cm**2)
+
+    # preallocate arrays
+    x_coordinates_ALL = np.random.uniform(0, 1, int(number_of_iterations))*no_xy
+    y_coordinates_ALL = np.random.uniform(0, 1, int(number_of_iterations))*no_xy
+    positive_array = np.zeros((no_xy, no_xy, no_z_with_buffer))
+    negative_array = np.zeros((no_xy, no_xy, no_z_with_buffer))
+    positive_array_temp = np.zeros((no_xy, no_xy, no_z_with_buffer))
+    negative_array_temp = np.zeros((no_xy,  no_xy, no_z_with_buffer))
+    no_recombined_charge_carriers = 0.0
+    no_initialised_charge_carriers = 0.0
+
+    '''
+    The following part is only used to plot the figure; consider to remove
```

Comment:
it doesn't make much sense to have matplotlib code inside numba parallelised function.